rig-bedrock/src/streaming.rs

```diff
@@ -0,0 +1,108 @@
+                        match delta {
+                            aws_bedrock::ContentBlockDelta::Text(text) => {
+                                if current_tool_call.is_none() {
+                                    yield Ok(StreamingChoice::Message(text))
+                                }
+                            },
+                            aws_bedrock::ContentBlockDelta::ToolUse(tool) => {
+                                if let Some(ref mut tool_call) = current_tool_call {
+                                    tool_call.input_json.push_str(tool.input());
+                                }
+                            },
+                            _ => {}
+                        }
+                    },
+                    aws_bedrock::ConverseStreamOutput::ContentBlockStart(event) => {
+                        match event.start.ok_or(CompletionError::ProviderError("ContentBlockStart has no data".into()))? {
+                            aws_bedrock::ContentBlockStart::ToolUse(tool_use) => {
+                                current_tool_call = Some(ToolCallState {
+                                    name: tool_use.name,
+                                    id: tool_use.tool_use_id,
+                                    input_json: String::new(),
+                                });
+                            },
+                            _ => yield Err(CompletionError::ProviderError("Stream is empty".into()))
+                        }
+                    },
+                    aws_bedrock::ConverseStreamOutput::ContentBlockStop(_) => {
+                        if let Some(tool_call) = current_tool_call.take() {
+                            let json_str = if tool_call.input_json.is_empty() {
+                                "{}"
+                            } else {
+                                &tool_call.input_json
+                            };
+                            match serde_json::from_str(json_str) {
+                                Ok(json_value) => {
+                                    yield Ok(StreamingChoice::ToolCall(
+                                        tool_call.name,
+                                        tool_call.id,
+                                        json_value,
+                                    ));
+                                },
+                                Err(e) => {
+                                    yield Err(CompletionError::from(e));
+                                }
+                            }
+
+                        }
+                    },
+                    // aws_bedrock::ConverseStreamOutput::MessageStart(message_start_event) => todo!(),
+                    // aws_bedrock::ConverseStreamOutput::MessageStop(message_stop_event) => todo!(),
+                    // aws_bedrock::ConverseStreamOutput::Metadata(converse_stream_metadata_event) => todo!(),
```

Comment:
We try not to leave commented blocks of code in final releases. Either leave a `TODO` comment explaining or cut these out!